### PR TITLE
Add common enum for debug info instructions from either opencl or vulkan

### PR DESCRIPTION
**Pull request by greg-lunarg**

### CODE REVIEW

File: BUILD.gn [hunk 1]
```diff
@@ -418,6 +418,7 @@ static_library("spvtools") {
     "source/binary.cpp",
     "source/binary.h",
     "source/cfa.h",
+    "source/common_debug_info.h.h",
```
Comment (dneto0) [inline BUILD.gn:421]: Looks like a typo with double .h

**Review by dneto0 — CHANGES_REQUESTED** [review]
Looks fine except for a typo in the BUILD.gn file.  (Sorry we don't seem to have a CI bot for that)

File: BUILD.gn (hunk 1 above)
Comment (greg-lunarg) [reply BUILD.gn:421]: Gah! Editing at 3am :( Good catch!

**Comment by greg-lunarg** [comment]
@jaebaek Please review

File: source/common_debug_info.h [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_COMMON_DEBUG_INFO_HEADER_H_
+#define SOURCE_COMMON_DEBUG_INFO_HEADER_H_
+
+// this enum defines the known common set of instructions that are the same
```
Comment (jaebaek) [inline source/common_debug_info.h:18]: // this enum defines ... --> // This enum defines

**Review by jaebaek — APPROVED** [review]
LGTM with a nit

File: source/opt/instruction.h [hunk 3]
```diff
@@ -550,10 +552,28 @@ class Instruction : public utils::IntrusiveNodeBase<Instruction> {
...
+  // Returns debug opcode of an OpenCL.100.DebugInfo or
+  // NonSemantic.Vulkan.DebugInfo.100 instruction. Since these overlap, we
+  // return the OpenCLDebugInfo code
+  CommonDebugInfoInstructions GetCommonDebugOpcode() const;
+
   // Returns true if it is an OpenCL.DebugInfo.100 instruction.
   bool IsOpenCL100DebugInstr() const {
     return GetOpenCL100DebugOpcode() != OpenCLDebugInfo100InstructionsMax;
   }
+  // Returns true if it is an NonSemantic.Vulkan.DebugInfo.100 instruction.
```
Comment (s-perron) [inline source/opt/instruction.h:569]: You have done this in a number of places.  Is "an NonSemantic instruction" correct?  I thought it should be "a NonSemantic instruction."

**Review by s-perron — CHANGES_REQUESTED** [review]
Generally looks good.  Just a few minor comments.

**Comment by greg-lunarg** [comment]
Thank you for the reviews! I believe I have addressed all concerns.

File: source/opt/instruction.h (hunk 3 above)
Comment (greg-lunarg) [reply source/opt/instruction.h]: Fixed

File: source/common_debug_info.h [hunk 4]
```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2017 Google Inc.
```
[the remark this answers is not in the document]
Comment (greg-lunarg) [reply source/common_debug_info.h]: Fixed

File: source/common_debug_info.h (hunk 2 above)
Comment (greg-lunarg) [reply source/common_debug_info.h]: Fixed

**Review by s-perron — APPROVED** [review]

**Pull request merged**